### ISSUE TITLE
issues when requesting hardware 

### ISSUE THREAD

**Issue by exodk** [snipe/snipe-it#3660]
#### Expected Behavior (or desired behavior if a feature request)
Go to request-able hardware 
able to request hardware
(what you expect to happen goes here)

-----

#### Actual Behavior
able to go to request-able hardware 
when click request get a message "Whoops, looks like something went wrong."
(what actually happens goes here)

-----

#### Please confirm you have done the following before posting your bug report:

- [yes ] I have enabled debug mode
- [yes ] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

-----
#### Please provide answers to these questions before posting your bug report:

- Version of Snipe-IT you're running - 3.6.6 build 1
- What OS and web server you're running Snipe-IT on - ubuntu server 14.04.1
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc) - install.sh
- WITH DEBUG TURNED ON, if you're getting an error in your browser, include that error

Whoops, looks like something went wrong.

1/1
Swift_RfcComplianceException in MailboxHeader.php line 347:
Address in mailbox given [] does not comply with RFC 2822, 3.6.2.
in MailboxHeader.php line 347
at Swift_Mime_Headers_MailboxHeader->_assertValidAddress('') in MailboxHeader.php line 263
at Swift_Mime_Headers_MailboxHeader->normalizeMailboxes(array('' => null)) in MailboxHeader.php line 106
at Swift_Mime_Headers_MailboxHeader->setNameAddresses(array('' => null)) in MailboxHeader.php line 63
at Swift_Mime_Headers_MailboxHeader->setFieldBodyModel(array('' => null)) in SimpleHeaderFactory.php line 58
at Swift_Mime_SimpleHeaderFactory->createMailboxHeader('Reply-To', array('' => null)) in SimpleHeaderSet.php line 68
at Swift_Mime_SimpleHeaderSet->addMailboxHeader('Reply-To', array('' => null)) in SimpleMessage.php line 268
at Swift_Mime_SimpleMessage->setReplyTo(array('' => null)) in SimpleMessage.php line 245
at Swift_Mime_SimpleMessage->addReplyTo(null, null) in Message.php line 137
at Message->addAddresses(null, null, 'ReplyTo') in Message.php line 121
at Message->replyTo(null, null) in ViewAssetsController.php line 173
at ViewAssetsController->App\Http\Controllers\{closure}(object(Message))
at call_user_func(object(Closure), object(Message)) in Mailer.php line 403
at Mailer->callMessageBuilder(object(Closure), object(Message)) in Mailer.php line 163
at Mailer->send('emails.asset-requested', array('item_quantity' => '1', 'asset_id' => '1', 'requested_date' => '2017-06-13 11:39:15', 'user_id' => '1', 'requested_by' => 'Jean-Paul Tager', 'item_name' => '', 'item_type' => 'asset', 'item_url' => 'http://192.168.8.15/hardware/1/view'), object(Closure)) in compiled.php line 6329
at Facade::__callStatic('send', array('emails.asset-requested', array('item_quantity' => '1', 'asset_id' => '1', 'requested_date' => '2017-06-13 11:39:15', 'user_id' => '1', 'requested_by' => 'Jean-Paul Tager', 'item_name' => '', 'item_type' => 'asset', 'item_url' => 'http://192.168.8.15/hardware/1/view'), object(Closure))) in ViewAssetsController.php line 175
at Mail::send('emails.asset-requested', array('item_quantity' => '1', 'asset_id' => '1', 'requested_date' => '2017-06-13 11:39:15', 'user_id' => '1', 'requested_by' => 'Jean-Paul Tager', 'item_name' => '', 'item_type' => 'asset', 'item_url' => 'http://192.168.8.15/hardware/1/view'), object(Closure)) in ViewAssetsController.php line 175
at ViewAssetsController->getRequestItem('asset', '1')
at call_user_func_array(array(object(ViewAssetsController), 'getRequestItem'), array('itemType' => 'asset', 'itemId' => '1')) in compiled.php line 9424
at Controller->callAction('getRequestItem', array('itemType' => 'asset', 'itemId' => '1')) in compiled.php line 9486
at ControllerDispatcher->call(object(ViewAssetsController), object(Route), 'getRequestItem') in compiled.php line 9466
at ControllerDispatcher->Illuminate\Routing\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 52
at Pipeline->Illuminate\Routing\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in compiled.php line 9948
at Pipeline->then(object(Closure)) in compiled.php line 9467
at ControllerDispatcher->callWithinStack(object(ViewAssetsController), object(Route), object(Request), 'getRequestItem') in compiled.php line 9454
at ControllerDispatcher->dispatch(object(Route), object(Request), 'App\Http\Controllers\ViewAssetsController', 'getRequestItem') in compiled.php line 8524
at Route->runController(object(Request)) in compiled.php line 8511
at Route->run(object(Request)) in compiled.php line 8225
at Router->Illuminate\Routing\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 52
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in Authenticate.php line 45
at Authenticate->handle(object(Request), object(Closure))
at call_user_func_array(array(object(Authenticate), 'handle'), array(object(Request), object(Closure))) in compiled.php line 9963
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in CheckForTwoFactor.php line 49
at CheckForTwoFactor->handle(object(Request), object(Closure))
at call_user_func_array(array(object(CheckForTwoFactor), 'handle'), array(object(Request), object(Closure))) in compiled.php line 9963
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in CheckLocale.php line 42
at CheckLocale->handle(object(Request), object(Closure))
at call_user_func_array(array(object(CheckLocale), 'handle'), array(object(Request), object(Closure))) in compiled.php line 9963
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in compiled.php line 3225
at VerifyCsrfToken->handle(object(Request), object(Closure))
at call_user_func_array(array(object(VerifyCsrfToken), 'handle'), array(object(Request), object(Closure))) in compiled.php line 9963
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in compiled.php line 13213
at AddQueuedCookiesToResponse->handle(object(Request), object(Closure))
at call_user_func_array(array(object(AddQueuedCookiesToResponse), 'handle'), array(object(Request), object(Closure))) in compiled.php line 9963
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in compiled.php line 13150
at EncryptCookies->handle(object(Request), object(Closure))
at call_user_func_array(array(object(EncryptCookies), 'handle'), array(object(Request), object(Closure))) in compiled.php line 9963
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in compiled.php line 9948
at Pipeline->then(object(Closure)) in compiled.php line 8226
at Router->runRouteWithinStack(object(Route), object(Request)) in compiled.php line 8217
at Router->dispatchToRoute(object(Request)) in compiled.php line 8207
at Router->dispatch(object(Request)) in compiled.php line 2419
at Kernel->Illuminate\Foundation\Http\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 52
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in Debugbar.php line 49
at Debugbar->handle(object(Request), object(Closure))
at call_user_func_array(array(object(Debugbar), 'handle'), array(object(Request), object(Closure))) in compiled.php line 9963
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in CheckForDebug.php line 25
at CheckForDebug->handle(object(Request), object(Closure))
at call_user_func_array(array(object(CheckForDebug), 'handle'), array(object(Request), object(Closure))) in compiled.php line 9963
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in TrustProxies.php line 46
at TrustProxies->handle(object(Request), object(Closure))
at call_user_func_array(array(object(TrustProxies), 'handle'), array(object(Request), object(Closure))) in compiled.php line 9963
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in CheckForSetup.php line 27
at CheckForSetup->handle(object(Request), object(Closure))
at call_user_func_array(array(object(CheckForSetup), 'handle'), array(object(Request), object(Closure))) in compiled.php line 9963
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in NosniffGuard.php line 17
at NosniffGuard->handle(object(Request), object(Closure))
at call_user_func_array(array(object(NosniffGuard), 'handle'), array(object(Request), object(Closure))) in compiled.php line 9963
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in XssProtectHeader.php line 17
at XssProtectHeader->handle(object(Request), object(Closure))
at call_user_func_array(array(object(XssProtectHeader), 'handle'), array(object(Request), object(Closure))) in compiled.php line 9963
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in FrameGuard.php line 17
at FrameGuard->handle(object(Request), object(Closure))
at call_user_func_array(array(object(FrameGuard), 'handle'), array(object(Request), object(Closure))) in compiled.php line 9963
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in compiled.php line 13474
at ShareErrorsFromSession->handle(object(Request), object(Closure))
at call_user_func_array(array(object(ShareErrorsFromSession), 'handle'), array(object(Request), object(Closure))) in compiled.php line 9963
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in compiled.php line 11964
at StartSession->handle(object(Request), object(Closure))
at call_user_func_array(array(object(StartSession), 'handle'), array(object(Request), object(Closure))) in compiled.php line 9963
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request)) in CheckForMaintenanceMode.php line 145
at CheckForMaintenanceMode->handle(object(Request), object(Closure))
at call_user_func_array(array(object(CheckForMaintenanceMode), 'handle'), array(object(Request), object(Closure))) in compiled.php line 9963
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in Pipeline.php line 32
at Pipeline->Illuminate\Routing\{closure}(object(Request))
at call_user_func(object(Closure), object(Request)) in compiled.php line 9948
at Pipeline->then(object(Closure)) in compiled.php line 2366
at Kernel->sendRequestThroughRouter(object(Request)) in compiled.php line 2350
at Kernel->handle(object(Request)) in index.php line 60

- What specific Snipe-IT page you're on, and what specific element you're interacting with to trigger the error
192.168.8.15/account/request/asset/1
- If a stacktrace is provided in the error, include that too.
- Any errors that appear in your browser's error console.
/account/request/asset/1:1 POST http://192.168.8.15/account/request/asset/1 500 (Internal Server Error)
- Confirm whether the error is [reproduceable on the demo](https://snipeitapp.com/demo).
no - works fine on demo
- Include any additional information you can find in `app/storage/logs` and your webserver's logs.
https://drive.google.com/open?id=0B_kNO06vSrMidFl4MkRBei1IV2s
- Include what you've done so far in the installation, and if you got any error messages along the way.
fresh install - added branding and 1 asset 
- Indicate whether or not you've manually edited any data directly in the database
have not manually touched database

**Comment by snipe** [comment]
`at Swift_Mime_Headers_MailboxHeader->_assertValidAddress('') in MailboxHeader.php line 263`

My guess would be the requester or the admin doesn't have a valid email address.

**Comment by exodk** [comment]
email addresses are fine, could it be lack of smtp sever?

**Comment by exodk** [comment]
could i just disable email alerts?

is there a page i can browse to that will show requests in order to be approved?

**Comment by exodk** [comment]
working now!

thank you for your time :)

**Comment by happicamper** [comment]
Sorry for this comment @snipe if this is a wrong place to answer or point me to the right place would be appreciated. I just want to share my solution with `RFC Swift_RfcComplianceException: Address in mailbox given [] does not comply with RFC 2822, 3.6.2` in Snipe IT Asset Management version v4.1.4 build 173 running in Docker. I've tried all of the Mail ENV parameters in this [documentation](https://snipe-it.readme.io/v3.6.2/docs/docker) and then it produces error `does not comply with RFC`. I've checked all the parameters needed in mail.php inside docker and these are:  
```
MAIL_DRIVER
MAIL_HOST
MAIL_PORT
MAIL_USERNAME
MAIL_PASSWORD
MAIL_ENCRYPTION
MAIL_FROM_ADDR
MAIL_FROM_NAME
MAIL_REPLYTO_ADDR
MAIL_REPLYTO_NAME
```
supplied all the variables in my ENV file and now it works!
